### PR TITLE
tfproviderlint: build with go@1.21

### PR DESCRIPTION
**Pull request by stefanb**
Moved from 
* https://github.com/Homebrew/homebrew-core/pull/162713

as a follow-up

Related:
* https://github.com/Homebrew/homebrew-core/pull/157782

----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
@stefanb might need some test update as well?

**Comment by stefanb** [comment]
It also needs to wait for 
* https://github.com/Homebrew/homebrew-core/pull/162713 

to be merged.

File: Formula/t/tfproviderlint.rb [hunk 1]
```diff
@@ -16,7 +17,8 @@ class Tfproviderlint < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "6d5103190a14b75826496cbe28614f49619a25d8f8492b4a1804e48e6c22bfca"
   end
 
-  depends_on "go" => [:build, :test]
+  # use "go" again when when https://github.com/bflad/tfproviderlint/pull/297 is released and released
+  depends_on "go@1.21" => [:build, :test]
```
Comment (chenrui333) [inline Formula/t/tfproviderlint.rb:21]: we can also just use go1.22 for test (working fine in my local)

File: Formula/t/tfproviderlint.rb [hunk 2]
```diff
@@ -29,6 +31,8 @@ def install
   end
 
   test do
+    ENV.prepend_path "PATH", Formula["go@1.21"].bin
+
```
Comment (chenrui333) [inline Formula/t/tfproviderlint.rb:35]: using `depends_on "go" => :test` would simplify test setup

File: Formula/t/tfproviderlint.rb [hunk 3]
```diff
@@ -4,6 +4,7 @@ class Tfproviderlint < Formula
   url "https://github.com/bflad/tfproviderlint/archive/refs/tags/v0.29.0.tar.gz"
   sha256 "60427ce6952106ba6c321555352645f051781ff55fe876b591e5dd2454852692"
   license "MPL-2.0"
+  revision 1
```
Comment (carlocab) [inline Formula/t/tfproviderlint.rb:7]: Do users need to install the new build, or will whatever they have installed previously still work?

If it's the latter we can omit the `revision` bump.

Comment (chenrui333) [reply Formula/t/tfproviderlint.rb:7]: yeah, I just tested, the existing binary works fine with the latest go

Comment (chenrui333) [reply Formula/t/tfproviderlint.rb]: also mark the PR as no-bottle build

**Review by p-linnane — APPROVED** [review]

**Pull request merged**